src/main/java/org/wildfly/security/auth/server/ServerAuthenticationContext.java

```diff
@@ -802,38 +794,6 @@ private void handleOne(final Callback[] callbacks, final int idx) throws IOExcep
                         }
                     }
                     handleOne(callbacks, idx + 1);
-                } else if (callback instanceof SSLSessionAuthorizationCallback) {
-                    final SSLSessionAuthorizationCallback authorizationCallback = (SSLSessionAuthorizationCallback) callback;
-                    final SSLSession sslSession = authorizationCallback.getSslSession();
-                    final IdentityCache cache = (IdentityCache) SSLUtils.computeIfAbsent(sslSession, "org.wildfly.elytron.identity-cache", key -> new IdentityCache());
-                    final SecurityDomain securityDomain = stateRef.get().getSecurityDomain();
-                    final SecurityIdentity identity = cache.identities.get(securityDomain);
```

Comment:
I didn't notice this requirement. That is the main part that I was looking for feedback.

Will see how we can do it using this generic solution.